### PR TITLE
Cherry-pick fixes from v1.3.x brach

### PR DESCRIPTION
**Pull request by Cyril-Engels**

### CODE REVIEW

File: src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java [hunk 1]
```diff
@@ -378,13 +404,25 @@ public String getTopic() {
     return topic;
   }
 
+  @SuppressWarnings("deprecated")
   public boolean logErrors() {
-    return !tolerateErrors() || getBoolean(ERRORS_LOG_ENABLE_CONFIG);
```
Comment (Cyril-Engels) [inline src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java:382]: Note: this was added recently from Mongo, which made a difference here 
https://github.com/confluentinc/mongo-kafka/pull/5/files#diff-cd6672a31761782e6b18f3e207148848f7950593821ef36dcad8d1872d570745L302-L307

**Comment by C0urante** [comment]
Have we tried to contribute these to the upstream repo? I filed https://jira.mongodb.org/browse/KAFKA-215, for example, to try to get the ball rolling on porting https://github.com/confluentinc/mongo-kafka/pull/4.

**Comment by Cyril-Engels** [comment]
> Have we tried to contribute these to the upstream repo? I filed https://jira.mongodb.org/browse/KAFKA-215, for example, to try to get the ball rolling on porting #4.

@C0urante I agree with you. We should definitely get it in to upstream repo to not diverge and prevent different behavior between on prem and cloud. But this should  not block our release of new version since it's already running in cloud. We can merge upstream back later.
I don't see any response in the JIRA https://jira.mongodb.org/browse/KAFKA-215. Do you want to open the PR first?

**Comment by C0urante** [comment]
> But this should not block our release of new version since it's already running in cloud. 

I'm a little confused--aren't we running 1.3 in cloud, not 1.5? And regardless, I think this _should_ block the release. We don't necessarily have to wait for things to get merged to the upstream but there at least need to be PRs opened. The more we accumulate fixes like this the harder it becomes to apply them due to merge conflicts and the likelihood that one gets dropped and causes an incident increases.

I plan to file a PR for KAFKA-215 sometime this week. You (or someone else) should do the same for the other two fixes here before we move forward with this release.

**Comment by C0urante** [comment]
Upstream pr for KAFKA-215 opened here: https://github.com/mongodb/mongo-kafka/pull/68

**Comment by Cyril-Engels** [comment]
@C0urante We have an email thread to discuss this fix https://github.com/confluentinc/mongo-kafka/pull/2, and what actions user can take. @ncliang opened a ticket https://jira.mongodb.org/browse/KAFKA-216, and will probably open PR. I will wait a bit.

**Review by C0urante — APPROVED** [review]
Radio silence from Mongo; we can merge now. Thanks for humoring me @Cyril-Engels

**Pull request merged**